### PR TITLE
Compatibility with php5.3 or higher

### PR DESCRIPTION
**Pull request by Berbass**
The minimum compatibility was previously set to php7 just because of one line of code, that I think is unfortunate.
This update fixes it.

### CODE REVIEW

**Comment by Webonaute** [comment]
please look at my comments.

thank.

**Comment by Berbass** [comment]
Hello @Webonaute sorry for for the delays on my answers.

Grosso modo, all I did was to remove the type hints of string scalar and replaced them with exception throws, as string type hint is not supported with php versions <= 7.

I've tested my mods and the bundle works well with php 5.5.9 and php 5.6.

File: Generator/DoctrineFixtureGenerator.php [hunk 1]
```diff
@@ -153,12 +153,12 @@ public function getFixtureNameFromEntityName($entity, array $ids = [], $prefix =
     {
         // Throw error if $entity is not a string
         if (!is_string($entity)) {
-            throw new \InvalidArgumentException('The parameter $entity is expected to be a string, "' . typeof($entity) . '" given.');
+            throw new \InvalidArgumentException('The parameter $entity is expected to be a string, "' . gettype($entity) . '" given.');
```
Comment (Berbass) [inline Generator/DoctrineFixtureGenerator.php]: typeof was a typo, as I'm used to use it in C programming, but it's not a php function.

File: Generator/DoctrineFixtureGenerator.php [hunk 2]
```diff
@@ -146,10 +146,21 @@ public function getFixtureFileName($entity, $name, array $ids)
...
      * @param string $prefix
+     * @throws InvalidArgumentException if the $entity or $prefix argument is not of type string
      * @return string
      */
-    public function getFixtureNameFromEntityName(string $entity, array $ids = [], string $prefix = null)
+    public function getFixtureNameFromEntityName($entity, array $ids = [], $prefix = null)
     {
+        // Throw error if $entity is not a string
+        if (!is_string($entity)) {
+            throw new \InvalidArgumentException('The parameter $entity is expected to be a string, "' . typeof($entity) . '" given.');
```
Comment (Berbass) [inline Generator/DoctrineFixtureGenerator.php]: I used gettype everywhere.

**Comment by Webonaute** [comment]
its pushed to php53-compatibility-unsupported branch.

**Pull request merged**